packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -51,115 +36,88 @@ export class ForceListMetadataExecutor extends SfdxCommandletExecutor<string> {
-    );
-    taskViewService.addCommandExecution(execution, cancellationTokenSource);
-  }
 }
 
-const workspaceChecker = new SfdxWorkspaceChecker();
-const parameterGatherer = new EmptyParametersGatherer();
-
 export async function forceListMetadata(
   metadataType: string,
-  defaultUsernameOrAlias: string
-) {
-  const outputPath = await getComponentsPath(
-    metadataType,
-    defaultUsernameOrAlias
-  );
-  const describeExecutor = new ForceListMetadataExecutor(
-    metadataType,
-    outputPath,
-    defaultUsernameOrAlias
-  );
-  const commandlet = new SfdxCommandlet(
-    workspaceChecker,
-    parameterGatherer,
-    describeExecutor
-  );
-  await commandlet.run();
+  defaultUsernameOrAlias: string,
+  outputPath: string
+): Promise<string> {
+  const execution = new CliCommandExecutor(
+    new ForceListMetadataExecutor(metadataType, defaultUsernameOrAlias).build(
+      {}
+    ),
+    { cwd: getRootWorkspacePath() }
+  ).execute();
+
+  const cmdOutput = new CommandOutput();
+  const result = await cmdOutput.getCmdResult(execution);
+  fs.writeFileSync(outputPath, result);
+  return result;
 }
 
 export async function getComponentsPath(
   metadataType: string,
   defaultUsernameOrAlias: string
 ): Promise<string> {
-  if (hasRootWorkspace()) {
-    try {
-      const workspaceRootPath = getRootWorkspacePath();
-      const username =
-        (await OrgAuthInfo.getUsername(defaultUsernameOrAlias)) ||
-        defaultUsernameOrAlias;
-
-      const componentsPath = path.join(
-        workspaceRootPath,
-        '.sfdx',
-        'orgs',
-        username,
-        'metadata',
-        metadataType + '.json'
-      );
-      return componentsPath;
-    } catch (e) {
-      telemetryService.sendError(e);
-      throw new Error(e);
-    }
-  } else {
+  if (!hasRootWorkspace()) {
     const err = nls.localize('cannot_determine_workspace');
     telemetryService.sendError(err);
     throw new Error(err);
   }
+
+  const workspaceRootPath = getRootWorkspacePath();
+
+  const username =
+    (await OrgAuthInfo.getUsername(defaultUsernameOrAlias)) ||
+    defaultUsernameOrAlias;
+
+  const componentsPath = path.join(
+    workspaceRootPath,
+    '.sfdx',
+    'orgs',
+    username,
+    'metadata',
+    metadataType + '.json'
+  );
+  return componentsPath;
 }
 
 export function buildComponentsList(
-  componentsPath: string,
-  metadataType: string
+  metadataType: string,
+  componentsFile?: string,
+  componentsPath?: string
 ): string[] {
-  try {
-    const metaComponents = [];
-    const fileData = fs.readFileSync(componentsPath, 'utf8');
-    if (fileData !== 'undefined') {
-      const fileObject = JSON.parse(fileData);
-      const cmpList = fileObject instanceof Array ? fileObject : [fileObject];
-      for (const component of cmpList) {
-        if (!isNullOrUndefined(component.fullName)) {
-          metaComponents.push(component.fullName);
-        }
+  const components = [];
+  if (isNullOrUndefined(componentsFile)) {
```

Comment:
`util.isNullOrUndefined` has been deprecated for awhile, we should avoid using it going forward.
https://nodejs.org/api/util.html#util_util_isnullorundefined_object

Reply:
Good catch